Guard functions that use types that aren't available on all platforms with
pre-processor conditions.

diff -urN readline-8.2/rlprivate.h.orig readline-8.2/rlprivate.h
--- readline-8.2/rlprivate.h.orig	2022-08-12 00:35:16.000000000 +0200
+++ readline-8.2/rlprivate.h	2022-11-12 17:43:54.266291863 +0100
@@ -303,7 +303,7 @@
 
 extern int _rl_timeout_init (void);
 extern int _rl_timeout_handle_sigalrm (void);
-#if defined (_POSIXSELECT_H_)
+#if defined (HAVE_PSELECT) || defined (HAVE_SELECT)
 /* use as a sentinel for fd_set, struct timeval,  and sigset_t definitions */
 extern int _rl_timeout_select (int, fd_set *, fd_set *, fd_set *, const struct timeval *, const sigset_t *);
 #endif
